src/controllers/assertionController.ts

```diff
@@ -0,0 +1,133 @@
+            } else {
+                let as = assertion.toJSON();
+                //make URL/ID absolute
+                as.id = tools.server_url + as.id;
+                res.json(as);
+            }
+        });
+};
+
+exports.assertion_create = function(req: Request, res: Response) {
+    let assertion = new Assertion(
+        {  "@context":  "https://w3id.org/openbadges/v2",
+        recipient: {type: "url", hashed: false, identity: req.body.receiver, name: req.body.receiverName},
+        sender: {identity: req.body.sender, name: req.body.senderName},
+        type: "Assertion",
+        badge: req.body.badgeclass,
+        issuedOn: new Date().toString(),
+        evidence: {id: req.body.reason },
+        verfication: { type: "hosted" }
+        });
+
+    assertion.save(function (err: Error) {
+        //TODO: is this error handling correct
+        if (err) { 
+            res.send(err)
+        } else {
+            res.status(200).send(tools.server_url + assertion.id);
+        }
+    });
+
+};
+
+//TODO: any type
+
+exports.assertion_accept =  function(req: Request, res: Response) {
+    Assertion.findByIdAndUpdate(req.params.id,
+        {$set: {accepted: true}},
+        {new: false})
+        .then((assertion: any) => { 
+            res.status(200).send(tools.server_url + assertion.id)
+        })
+        .catch((err: Error) => {
+            res.send(err)
+        });
+};
+
+exports.assertion_delete = function(req: Request, res: Response) {
+    Assertion.findByIdAndDelete(req.params.id, (err: Error, docs: any) => { 
+        if (err){ 
+            res.send(err);
+        } else { 
+            res.status(200).send();
+        } 
+    });
+};
+
+// refactor needed D:
+exports.assertion_badge = function(req: Request, res: Response) {
+    Assertion.findById(req.params.id)
+    .exec(function (err: Error, assertion: any) {
+        if (assertion == null) {
+            res.status(404).send();
+            return;
+        }
+        Badgeclass.findById(assertion.badge.split('/').pop())
+        .exec(function (err: Error, badgeclass: any) {
+            if (badgeclass == null){
+                res.status(404).send()
+                return;
+            }
+            async.waterfall([
+                async.apply(getBadgeImage, badgeclass.image)
+            ], function (err: Error, badgeImage: any) {
+                if (badgeImage == null)
+                    return;
+
+                bakery.bake({
+                    image: badgeImage,
+                    assertion: assertion.toJSON()}, 
+                    function (err: Error, imageData: any) {
+                        res.set('Content-Type', 'image/png');
+                        res.set('Content-Disposition', 'attachment; filename='+ badgeclass.name + '.png');
+                        res.set('Content-Length', imageData.length);
+                        res.end(imageData, 'binary');
+                        return;
+                });
+            });
```

Comment:
Abstracting this stuff out to a separately declared declared will go a long way toward your refactor :)